airbyte-integrations/bases/base-typing-deduping/src/test/java/io/airbyte/integrations/base/destination/typing_deduping/AirbyteTypeTest.java

```diff
@@ -451,4 +452,51 @@ public void testChooseOneOf() {
     assertEquals(AirbyteProtocolType.STRING, AirbyteTypeUtils.chooseOneOfType(o));
   }
 
+  @Test
+  public void testAsColumns() {
+    OneOf o = new OneOf(List.of(
+        AirbyteProtocolType.STRING,
+        new Struct(new LinkedHashMap<>() {{
+          put("foo", AirbyteProtocolType.STRING);
+        }}),
+        new Array(AirbyteProtocolType.STRING),
+        // This is bad behavior, but it matches current behavior so we'll test it.
+        // Ideally, we would recognize that the sub-oneOfs are also objects.
+        new OneOf(List.of(new Struct(new LinkedHashMap<>()))),
```

Comment:
I guess technically this is also currently impossible behavior, because there's no way to generate nested OneOfs (only nested UnsupportedOneOf). :shrug: